contrib/offline/manage-offline-container-images.sh

```diff
@@ -44,20 +44,9 @@ function create_container_image_tar() {
 		set -e
 		sudo docker save -o ${FILE_NAME}  ${image}
 
-		# NOTE: Here removes the following repo parts from each image
+		# NOTE: Here removes the first part from each image
 		# so that these parts will be replaced with Kubespray.
-		# - kube_image_repo: "k8s.gcr.io"
-		# - gcr_image_repo: "gcr.io"
-		# - docker_image_repo: "docker.io"
-		# - quay_image_repo: "quay.io"
-		FIRST_PART=$(echo ${image} | awk -F"/" '{print $1}')
-		if [ "${FIRST_PART}" = "k8s.gcr.io" ] ||
-		   [ "${FIRST_PART}" = "gcr.io" ] ||
-		   [ "${FIRST_PART}" = "docker.io" ] ||
-		   [ "${FIRST_PART}" = "quay.io" ]; then
-			image=$(echo ${image} | sed s@"${FIRST_PART}/"@@)
-		fi
-		echo "${FILE_NAME}  ${image}" >> ${IMAGE_LIST}
+		echo "${FILE_NAME}  ${image#*/}" >> ${IMAGE_LIST}
```

Comment:
Which official repo don't we include in this list that you need a generic approach like you propose?

Reply:
I guess the reason is

> it doesn't seem to work properly for registering some private registry images to the local registry.

from the pull request message.
@ErikJiang might want to replace
`dce-10-6-170-10:5000/10.6.182.20:5000/calico/cni:v3.19.2` to
`10.6.182.20:5000/calico/cni:v3.19.2`
for example.

I think it is better to have some option to specify a private registry instead of replacing the above.
For example, here can be changed to
```
		if [ "${FIRST_PART}" = "k8s.gcr.io" ] ||
		   [ "${FIRST_PART}" = "gcr.io" ] ||
		   [ "${FIRST_PART}" = "docker.io" ] ||
		   [ "${FIRST_PART}" = "quay.io" ] ||
		   [ "${FIRST_PART}" = "${PRIVATE_REGISTRY}" ]; then
```
by having a new option PRIVATE_REGISTRY.

Reply:
Indeed this would be a better approach.